### ISSUE TITLE
route.query is not a function

### ISSUE THREAD

**Issue by Ai01** [4Catalyzer/found-relay#54]
in found-relay@0.3.0-alpha.4 graphql result is a object:
![2017-07-20 1 41 10](https://user-images.githubusercontent.com/18245929/28402191-21669b9e-6d51-11e7-9fb0-d9265027d21b.png)
but in relay modern SnapshotRenderer class render function match.route.query neet to be a function.
![2017-07-20 1 42 05](https://user-images.githubusercontent.com/18245929/28402298-b186b8e4-6d51-11e7-97de-35eac59b0a17.png)
<img width="271" alt="2017-07-20 1 44 06" src="https://user-images.githubusercontent.com/18245929/28402432-6a9157b8-6d52-11e7-8b04-8caba756f10d.png">

after change to the found-relay@0.3.0-alpha.1 it's ok

**Comment by taion** [comment]
Well, does the query actually work? How are you hitting this? Compiled queries are usually always functions...

**Comment by rojobuffalo** [comment]
Is it ok to have a `Route` that doesn't have a `query` prop? Like for a static component. I'm getting the same error but I'm not sure which Route it's coming from.
```
<Route Component={HomePage} />
```

**Comment by taion** [comment]
Yes – this error could only come up if the route query is somehow malformed, but I don't see how that can normally happen.

**Comment by rojobuffalo** [comment]
I'm currently wrestling it so if you have any ideas about lines of attack lmk.

**Comment by taion** [comment]
Set a breakpoint and see what your query actually is, and what that route is?

**Comment by rojobuffalo** [comment]
Ok, while I do that, this is what the code looks like:

* Home page and not found render as expected. And the query works when I test it on GraphiQL.

## `routes.js`
```
export default makeRouteConfig(
  <Route path={'/'} Component={CorePage}>
    <Route Component={HomePage} />
    <Route
      path={'user/:userId'}
      Component={UserContainer}
      query={UserContainerQuery}
    />
    <Route path={'*'} Component={NotFound} />
  </Route>,
);
```

## `UserContainerQuery.js`
```
import { graphql } from 'react-relay';

const UserContainerQuery = graphql`
  query UserContainerQuery($userId: Uuid!) {
    userByRowId(rowId: $userId) {
      ...UserContainer_userByRowId
    }
  }
`;

export default UserContainerQuery;
```

## `UserContainer.js`
```
const UserContainer = props => <TransitionWrapper>
  <Layout page>
    <Menu
      baseUrl={`/user/${props.userByRowId.rowId}`}
      header={{icon: <PersonIcon />, title: 'User'}}
      disabled
    />
  </Layout>
</TransitionWrapper>;

export default createFragmentContainer(
  UserContainer,
  graphql`
    fragment UserContainer_userByRowId on User {
      rowId,
    }
  `,
);
```

**Comment by rojobuffalo** [comment]
I think I see the problem. I didn't setup the Relay `Network` to hit my API.

**Comment by rojobuffalo** [comment]
In the docs it looks like `Network.create` takes a function as a parameter. But in your example you pass `{schema}` which is your mock-api. But I don't see docs on passing an options object into the constructor. 

https://facebook.github.io/relay/docs/network-layer.html#network-layer

**Comment by rojobuffalo** [comment]
Oh, sorry. I see you're using `relay-local-schema`. That's the problem.

**Comment by rojobuffalo** [comment]
These are the args getting passed to `fetch`

```
{  
  "fragment":{  
    "argumentDefinitions":[  
      {  
        "kind":"LocalArgument",
        "name":"userId",
        "type":"Uuid!",
        "defaultValue":null
      }
    ],
    "kind":"Fragment",
    "metadata":null,
    "name":"UserContainerQuery",
    "selections":[  
      {  
        "kind":"LinkedField",
        "alias":null,
        "args":[  
          {  
            "kind":"Variable",
            "name":"rowId",
            "variableName":"userId",
            "type":"Uuid!"
          }
        ],
        "concreteType":"User",
        "name":"userByRowId",
        "plural":false,
        "selections":[  
          {  
            "kind":"FragmentSpread",
            "name":"UserContainer_userByRowId",
            "args":null
          }
        ],
        "storageKey":null
      }
    ],
    "type":"Query"
  },
  "id":null,
  "kind":"Batch",
  "metadata":{  

  },
  "name":"UserContainerQuery",
  "query":{  
    "argumentDefinitions":[  
      {  
        "kind":"LocalArgument",
        "name":"userId",
        "type":"Uuid!",
        "defaultValue":null
      }
    ],
    "kind":"Root",
    "name":"UserContainerQuery",
    "operation":"query",
    "selections":[  
      {  
        "kind":"LinkedField",
        "alias":null,
        "args":[  
          {  
            "kind":"Variable",
            "name":"rowId",
            "variableName":"userId",
            "type":"Uuid!"
          }
        ],
        "concreteType":"User",
        "name":"userByRowId",
        "plural":false,
        "selections":[  
          {  
            "kind":"ScalarField",
            "alias":null,
            "args":null,
            "name":"rowId",
            "storageKey":null
          },
          {  
            "kind":"ScalarField",
            "alias":null,
            "args":null,
            "name":"id",
            "storageKey":null
          }
        ],
        "storageKey":null
      }
    ]
  },
  "text":"query UserContainerQuery(\n  $userId: Uuid!\n) {\n  userByRowId(rowId: $userId) {\n    ...UserContainer_userByRowId\n    id\n  }\n}\n\nfragment UserContainer_userByRowId on User {\n  rowId\n}\n"
}
```

**Comment by rojobuffalo** [comment]
The response looks successful...

```
{  
  "data":{  
    "userByRowId":{  
      "rowId":"d58a6213-7c48-4a54-9ae7-17427c0dd679",
      "id":"WyJ1c2VycyIsImQ1OGE2MjEzLTdjNDgtNGE1NC05YWU3LTE3NDI3YzBkZDY3OSJd"
    }
  }
}
```

**Comment by rojobuffalo** [comment]
This error also shows up right before the `route.query is not a function` error.
```
TypeError: Cannot read property '__reactInternalInstance$uo5ztxqbex' of null
```

**Comment by rojobuffalo** [comment]
It looks like `props.route.query` is an object with two properties that are functions `classic()` and `modern()`. Is that where it is expecting `query` to be a function?

**Comment by rojobuffalo** [comment]
Sorry for polluting the thread with all that stuff that didn't apply to this issue. Just want to confirm that rolling back to `v0.3.0-alpha.1` made this issue go away.

**Comment by taion** [comment]
Ah, I forgot about compat. I'll see.

**Comment by taion** [comment]
Fixed in [v0.3.0-alpha.5](https://github.com/4Catalyzer/found-relay/releases/tag/v0.3.0-alpha.5).
